static/js/containers/PostPage.js

```diff
@@ -107,23 +66,45 @@ class PostPage extends React.Component {
             <PostDisplay post={post} toggleUpvote={toggleUpvote(dispatch)} expanded />
             <ReplyToPostForm forms={forms} post={post} />
           </Card>
-          <CommentTree comments={commentTreeData} forms={forms} upvote={this.upvote} downvote={this.downvote} />
+          <CommentTree comments={commentsTree} forms={forms} upvote={this.upvote} downvote={this.downvote} />
         </div>
       </div>
     )
   }
+}
 
-  render() {
-    const { posts, channels } = this.props
-    return <Loading restStates={[posts, channels]} renderContents={this.renderContents} />
+const shouldUpdateRequirements = R.complement(R.allPass([R.eqProps("postID"), R.eqProps("channelName")]))
+
+const updateRequirements = ({ dispatch, channelName, postID, channel }) => {
+  if (!postID || !channelName) {
+    // should not happen, this should be guaranteed by react-router
+    throw Error("Match error")
```

Comment:
well, if I throw the error then I can't redirect and vice versa. I'm inclined to leave it as-is for now and file a separate issue for Sentry integration linking to this line as one to fix.